main.py

```diff
@@ -115,41 +118,29 @@ def setup_target_task_training(args, target_tasks, model, strict):
 
     if args.load_target_train_checkpoint not in ("none", ""):
         # This is to load a particular target train checkpoint.
-        checkpoint = glob.glob(args.load_target_train_checkpoint)
-        assert len(checkpoint) > 0, (
-            "Specified load_target_train_checkpoint not found: %r"
-            % args.load_target_train_checkpoint
-        )
-        assert len(checkpoint) == 1, (
-            "Too many checkpoints match pattern: %r" % args.load_target_train_checkpoint
-        )
-        best_path = checkpoint[0]
-        log.info("Loading existing model from %r...", best_path)
-        load_model_state(model, best_path, args.cuda, task_names_to_avoid_loading, strict=strict)
+        model_path = get_best_checkpoint_path(args, "target_train")
+        log.info("Loading existing model from %r...", model_path)
+        load_model_state(model, model_path, args.cuda, task_names_to_avoid_loading, strict=strict)
```

Comment:
Why do we have the actual load happen both here and in reload? Why not just do setup once, and then reload at the start of training for each task.

Reply:
That is exactly what is being done here - we do setup once, and then we reload at the start of each task. We don't reload at index = 0 since we already do that once with setup. 
Actually, I just realized that we could just avoid loading in the setup_target_tasks function and just keep it in reload_tasks

Reply:
That would make the code much simpler and there wouldn't be the weird index = 0 type logic